enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/integrations/kernel/FulltextIndexAccessor.java

```diff
@@ -0,0 +1,226 @@
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.api.impl.fulltext.integrations.kernel;
+
+import org.apache.lucene.document.Document;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Iterator;
+
+import org.neo4j.graphdb.ResourceIterator;
+import org.neo4j.helpers.collection.BoundedIterable;
+import org.neo4j.io.pagecache.IOLimiter;
+import org.neo4j.kernel.api.exceptions.index.IndexEntryConflictException;
+import org.neo4j.kernel.api.impl.fulltext.lucene.ScoreEntityIterator;
+import org.neo4j.kernel.api.impl.fulltext.lucene.LuceneFulltextDocumentStructure;
+import org.neo4j.kernel.api.impl.fulltext.lucene.ReadOnlyFulltext;
+import org.neo4j.kernel.api.impl.fulltext.lucene.WritableFulltext;
+import org.neo4j.kernel.api.impl.schema.reader.LuceneAllEntriesIndexAccessorReader;
+import org.neo4j.kernel.api.index.IndexAccessor;
+import org.neo4j.kernel.api.index.IndexEntryUpdate;
+import org.neo4j.kernel.api.index.IndexUpdater;
+import org.neo4j.kernel.api.index.PropertyAccessor;
+import org.neo4j.kernel.impl.api.index.IndexUpdateMode;
+import org.neo4j.storageengine.api.schema.IndexReader;
+import org.neo4j.values.storable.Value;
+
+import static java.util.Collections.emptyIterator;
+
+public class FulltextIndexAccessor implements IndexAccessor
+{
+    private WritableFulltext luceneFulltext;
+    private FulltextIndexDescriptor descriptor;
+
+    FulltextIndexAccessor( WritableFulltext luceneFulltext, FulltextIndexDescriptor descriptor )
+    {
+        this.luceneFulltext = luceneFulltext;
+        this.descriptor = descriptor;
+    }
+
+    @Override
+    public void drop() throws IOException
+    {
+        luceneFulltext.drop();
+    }
+
+    @Override
+    public IndexUpdater newUpdater( IndexUpdateMode mode )
+    {
+        if ( luceneFulltext.isReadOnly() )
+        {
+            throw new UnsupportedOperationException( "Can't create updater for read only index." );
+        }
+        return new FulltextIndexUpdater( mode.requiresIdempotency(), mode.requiresRefresh() );
+    }
+
+    @Override
+    public void force( IOLimiter ioLimiter ) throws IOException
+    {
+        luceneFulltext.markAsOnline();
+        luceneFulltext.maybeRefreshBlocking();
+    }
+
+    @Override
+    public void refresh() throws IOException
+    {
+        luceneFulltext.maybeRefreshBlocking();
+    }
+
+    @Override
+    public void close() throws IOException
+    {
+        luceneFulltext.close();
+    }
+
+    @Override
+    public IndexReader newReader()
+    {
+        return IndexReader.EMPTY;
+    }
+
+    @Override
+    public BoundedIterable<Long> newAllEntriesReader()
+    {
+        //TODO support consistency check at some point
+        BoundedIterable<Document> empty = new BoundedIterable<Document>()
+        {
+            @Override
+            public long maxCount()
```

Comment:
todo

Reply:
Yeah, consistency checker isn't supported yet, and this is to keep track of where the rough edges are.

Reply:
This will be seen to when we add consistency checker support.